libraries/Microsoft.Bot.StreamingExtensions/IContentStream.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Implemented by stream attachments compatible with the Bot Framework Protocol 3 with Streaming Extensions.
+    /// </summary>
+    public interface IContentStream
+    {
+        /// <summary>
+        /// Gets a guid to use as the unique identifier of this ContentStream.
+        /// </summary>
+        /// <value>
+        /// Guid.
+        /// </value>
+        Guid Id { get; }
+
```

Comment:
Why are we requiring the uniqe id to be a guid? Wouldn't "1" or "2" or "Stream1" or other similar easy-to-read value work? Why do these need to be globally unique?

Reply:
Stream IDs end up being the identifier for the assemblers building them as they come in, the less unique they are the more compound the assembler IDs will become. Also they have to fit into the same ID space in the Header as the other message types.